requests_toolbelt/adapters/appengine.py

```diff
@@ -44,6 +49,57 @@ def __init__(self, validate_certificate=True, *args, **kwargs):
         super(AppEngineAdapter, self).__init__(self, *args, **kwargs)
 
     def init_poolmanager(self, connections, maxsize, block=False):
-        self.poolmanager = gaecontrib.AppEngineManager(
-            validate_certificate=self._validate_certificate
-        )
+        self.poolmanager = _AEPoolManager(self._validate_certificate)
+
+
+class _AEPoolManager(object):
+    """Implements urllib3's PoolManager API."""
+
+    def __init__(self, validate_certificate=True):
+        self.validate_certificate = validate_certificate
+
+    def connection_from_url(self, url):
+        return _AEConnection(self, url)
```

Comment:
No, the connection needs to be specific to the url itself:

This url contains the full path including hostname. The AEConnection's urlopen() is called later, but it is called with a host-relative URL that is missing the hostname. The only way to get the hostname is from the connection, which was initialized with the url-including-hostname. So that's my argument as to why I cannot reuse these...unless I'm missing something?